### PR TITLE
test(e2e): introduce web components build script for isolated components

### PR DESCRIPTION
**Pull request by jeffchew**
### Related Ticket(s)

#6250 

### Description

This PR will remove the older integration bundle build tests, and replace with a new script that will build all code sandbox examples. 

In addition, a CI flow has been drafted, and will be updated to point to the main repo once this is merged in.

There is a follow-on issue #6251 that will add cypress to testing each individual component based on the built isolated components.

### Changelog

**New**

- e2e test build script for web components

**Removed**

- original integration bundle/build script

### CODE REVIEW

File: packages/web-components/tests/e2e/build/replace-dependencies.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+const fs = require('fs');
+const program = require('commander');
+
+program.option('-f, --folder <folder>', 'Location of packages folder');
+program.parse();
+
+/**
+ * Command line options
+ *
+ * @type {commander.OptionValues}
```
Comment (kennylam) [inline packages/web-components/tests/e2e/build/replace-dependencies.js:21]:
```suggestion
 * @type {program.OptionValues}
```

File: packages/web-components/tests/e2e/build/build-examples.js [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+program
+  .option('-C, --skip-clean', 'Skips cleaning build folders')
+  .option('-P, --skip-packages', 'Skips the local package setup')
+  .option('-E, --skip-examples', 'Skips the local examples setup');
+program.parse();
+
+/**
+ * Command line options
+ *
+ * @type {commander.OptionValues}
```
Comment (kennylam) [inline packages/web-components/tests/e2e/build/build-examples.js:26]:
```suggestion
 * @type {program.OptionValues}
```

File: packages/web-components/gulp-tasks/config.js [hunk 3]
```diff
@@ -26,7 +27,9 @@ const { browser: browsers, spec: specs, ...rest } = commander
   .option('--update-snapshot', 'Updates snapshot')
   .option('--verbose', 'Enables verbose output')
   .parse(process.argv);
-const cloptions = { browsers: Array.from(browsers), specs: Array.from(specs), ...rest };
+
+// const cloptions = { browsers: Array.from(browsers), specs: Array.from(specs), ...rest };
```
Comment (kennylam) [inline packages/web-components/gulp-tasks/config.js:31]: Looks like you forgot to remove this.

File: packages/web-components/tests/e2e/build/build-examples.js [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+    Object.keys(_packages).forEach(pack => {
+      log(chalk.green(`Building package: ${pack}`));
+
+      const commands = [];
+      commands.push(
+        `cd "${_packages[pack]}" && yarn pack --filename "${_localPackagesFolder}/carbon-ibmdotcom-${pack}.tar.gz"`,
+        `tar xzf "${_localPackagesFolder}/carbon-ibmdotcom-${pack}.tar.gz" --directory "${_localPackagesFolder}"`,
+        `mv "${_localPackagesFolder}/package" "${_localPackagesFolder}/ibmdotcom-${pack}"`,
+        // eslint-disable-next-line max-len
+        `node "${_testScriptFolder}/replace-dependencies.js" -f "${_localPackagesFolder}" "${_localPackagesFolder}/ibmdotcom-${pack}/package.json"`
```
Comment (kennylam) [inline packages/web-components/tests/e2e/build/build-examples.js:134]: Just a comment but we can probably test this out later and see if skipping the decompression and linking the tarball directly into the `package.json` could maybe squeeze out a little bit of performance.

**Review by kennylam — COMMENTED** [review]
Thanks @jeffchew! Just a few small clean-up issues that GH annotation is complaining about.

File: packages/web-components/tests/e2e/build/build-examples.js (hunk 4 above)
Comment (jeffchew) [reply packages/web-components/tests/e2e/build/build-examples.js:134]: sure, most of this was pre-existing and largely kept this part the same other than different paths.

File: packages/web-components/gulp-tasks/config.js (hunk 3 above)
Comment (jeffchew) [reply packages/web-components/gulp-tasks/config.js]: thanks, got rid of it!

**Review by kennylam — APPROVED** [review]
LGTM thanks @jeffchew!

**Pull request merged**